agent/engine/dockerstate/docker_task_engine_state.go

```diff
@@ -102,6 +108,8 @@ type DockerTaskEngineState struct {
 	ipToTask               map[string]string // ip address -> task arn
 	v3EndpointIDToTask     map[string]string // container's v3 endpoint id -> taskarn
 	v3EndpointIDToDockerID map[string]string // container's v3 endpoint id -> DockerId
+
+	client api.ECSClient
```

Comment:
I agree it seems out of place. But the task metadata server is not a struct and is initialized through static methods. The alternatives here would be: 
- pass `client` as a parameter through the chain of calls just like `state` 
- initialize a global variable in an `init()` function within task metadata server

Because we already pass state through the chain of calls, I thought it best to place the client within the state, where it can be maintained.

Reply:
@haikuoliu Refactored to pass `client` as a parameter through call chain.